### ISSUE TITLE
Custom Semiology freezes Slicer SVT without any error messages, while "Setting Colours..."

### ISSUE THREAD

**Issue by thenineteen** [thenineteen/Semiology-Visualisation-Tool#135]
I've added a term `awesome` to the current version of the excel SemioBrain database. 

When adding awesome as a semiology, it hangs. We can `Cancel` the query, but it otherwise gets stuck at 11% as below. Other custom semiologies work ok.

![image](https://user-images.githubusercontent.com/10786099/86836054-dafe1980-c094-11ea-807e-32056e5c25df.png)

**Comment by thenineteen** [comment]
I thought maybe it may be due to settings such as the `Min 2D opacity` under visualisation (as it should only return 1 region and so maybe can't make transparent) - but no.

Also searching for custom terms such as `TL0` and `TL` (devised to calibrate the mappings and solve #75 and #82 ) only works for `TL` but not for `TL0`. `TL` returns a host of regions, whereas TL followed by a number should return less regions (or sometimes even just 1 region, as in the case of TL0)

so it makes me think maybe it is an issue with only 1 semiology 1 region?! I don't know. Also can't seem to debug it as it hangs my vs code too

**Comment by thenineteen** [comment]
I've got it to work now 

```
patient  = Semiology('awesome',
    Laterality.NEUTRAL,
    Laterality.NEUTRAL,
)

heatmap = patient.get_num_datapoints_dict()
print('Result:', heatmap)
```


returns

`Result: {103: 1.0, 104: 1.0, 173: 1.0, 174: 1.0}`

**Comment by fepegar** [comment]
I do get an error message when I query for awesome neutral:

<img width="532" alt="Screen Shot 2020-07-07 at 21 24 41" src="https://user-images.githubusercontent.com/12688084/86838664-5ad9b300-c098-11ea-9b4b-cae6963e8dea.png">

**Comment by fepegar** [comment]
Same for `TLb`.

**Comment by thenineteen** [comment]
hmm I still don't... let me try again, again tomorrow...

**Comment by fepegar** [comment]
I tried again today after pulling again and I do get an error in the console, I'll create an issue:

```python-traceback
RuntimeWarning: invalid value encountered in double_scalars
  normalizedScore /= (maxScore - minScore)
Traceback (most recent call last):
  File "/Users/fernando/git/Semiology-Visualisation-Tool/slicer/SemiologyVisualisation.py", line 666, in updateColors
    min2dOpacity=self.min2dOpacitySlider.value,
  File "/Users/fernando/git/Semiology-Visualisation-Tool/slicer/SemiologyVisualisation.py", line 1262, in setScoresColors
    color = self.getColorFromScore(normalizedScore, colorNode)
  File "/Users/fernando/git/Semiology-Visualisation-Tool/slicer/SemiologyVisualisation.py", line 1283, in getColorFromScore
    scoreIndex = int(round((numColors - 1) * normalizedScore))
ValueError: cannot convert float NaN to integer
```

**Comment by fepegar** [comment]
Why does "awesome" work anyway?

**Comment by thenineteen** [comment]
Because I added it as a test, because my calibration semiologies weren't working
